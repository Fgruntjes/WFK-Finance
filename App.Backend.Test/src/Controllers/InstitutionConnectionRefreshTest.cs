namespace App.Backend.Test.Controllers;

public class InstitutionConnectionRefreshTest : IClassFixture<InstitutionConnectionRefreshFixture>
{
	private readonly InstitutionConnectionRefreshFixture _fixture;

	public InstitutionConnectionRefreshTest(InstitutionConnectionRefreshFixture fixture)
	{
		_fixture = fixture;
	}

	[Fact]
	public async Task ByExternalId_Success()
	{
		var result = await _fixture.ExecuteQuery(new
		{
			_fixture.InstitutionConnectionEntity.ExternalId
		});
		result.MatchSnapshot();
	}

	[Fact]
	public async Task ByExternalId_MissingConnection()
	{
		var result = await _fixture.ExecuteQuery(new
		{
			ExternalId = "SomeExternalIdMissing"
		});
		result.MatchSnapshot(m => m.IgnoreField("errors[0].extensions.timestamp"));
	}

	[Fact]
	public async Task ById_Success()
	{
		var result = await _fixture.ExecuteQuery(new
		{
			_fixture.InstitutionConnectionEntity.Id
		});
		result.MatchSnapshot();

		// Assert database
		var connectionEntity = await _fixture.Database.InstitutionConnections
			.FindAsync(_fixture.InstitutionConnectionEntity.Id);
		connectionEntity.Should().NotBeNull();
		connectionEntity?.Accounts.Should().HaveCount(2);
	}

	[Fact]
	public async Task ById_MissingConnection()
	{
		var result = await _fixture.ExecuteQuery(new
		{
			Id = new Guid("59a35c45-6e8d-4dc7-bacc-f151f94da93d")
		});
		result.MatchSnapshot(m => m.IgnoreField("errors[0].extensions.timestamp"));
	}
}