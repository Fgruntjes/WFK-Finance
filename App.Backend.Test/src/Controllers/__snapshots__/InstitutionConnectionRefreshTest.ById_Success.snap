{
  "data": {
    "institutionConnection": {
      "refreshId": {
        "externalId": "ed69f988-a1fb-4e89-8d56-66b42e43a675",
        "connectUrl": "https://www.example.com/connect-url/refresh",
      }
    }
  }
}
